statistics/handle/handle.go

```diff
@@ -1010,7 +1016,22 @@ func (h *Handle) statsCacheLen() int {
 // updateStatsCache overrides the global statsCache with a new one, it may fail
 // if the global statsCache has been modified by others already.
 // Callers should add retry loop if necessary.
-func (h *Handle) updateStatsCache(newCache *cache.StatsCache) (updated bool) {
+func (h *Handle) updateStatsCache(tables []*statistics.Table, deletedIDs []int64, opts ...cache.TableStatsOpt) (updated bool) {
+	var maxVersion uint64
+	for _, tbl := range tables {
+		if tbl.Version > maxVersion {
+			maxVersion = tbl.Version
+		}
+	}
+	for currentVer := h.latestUpdateTableVersion.Load(); maxVersion > currentVer; {
+		if h.latestUpdateTableVersion.CompareAndSwap(currentVer, maxVersion) {
+			break
+		}
+		currentVer = h.latestUpdateTableVersion.Load()
+	}
+
+	oldCache := h.statsCache.Load()
+	newCache := oldCache.CopyAndUpdate(tables, deletedIDs, opts...)
```

Comment:
The update of version and cache pointer is not atomic. For example, both A and B call updateStatsCache, and the execution order is A.updateVersion->B.updateVersion->B.updateCache->A.updateCache.